api/game/clueless.py

```diff
@@ -24,56 +29,226 @@ def __init__(self, *args, **kwargs):
         ]
         self.secret_passages = ['study_kitchen', 'lounge_conservatory']
 
+        # TODO connection manager should contain each client's character choice
+          # self.players should be filled & board should be initialized 
+          # after all clients have connected and game starts. 
+        
         # players dict holds client_id: character assignments
         self.players = {}
         self.state = self.initialize_board(self.players)
+    
+
+    def jsonify(self):
+        return json.dumps(self.state)
+
+
+    def get_game_state(self):
+        return self.state
+
 
-    def initialize_board(self, players):
-        """ Places characters in starting locations, generates sceanario, 
+    def initialize_board(self, players: Dict) -> Dict:
+        """ Places characters in starting locations, generates scenario, 
         distributes cards """
 
         # data structs are just placeholders; can/should be changed
         state = {
             'suspect_locations': {},  # dict of suspect: room/hallway loc
             'concealed_scenario': {},
-            'player_cards': {},  # dict of player: player's card list 
+            'player_cards': {},  # dict of client id: player's card list 
             'visible_cards': [],  # list of cards shown to all players
             'turn_order': [],  # player turn order
-            'current_turn': 0,  # whose turn is it
+            'current_turn': (),  # whose turn is it? (client id, charname)

```

Comment:
can just be char name, make an enum for it or something

connection manager stores the lookup

game logic should not understand the concept of "client ids"

Reply:
actually, remove both this and turn order above, make turn order a deque, and just keep rotating the list in place

Reply:
opted to leave turn order as a list for now. What's best here depends on how we want to handle disproving suggestions, which isn't implemented yet (its a client-client interaction that only changes the game state at the beginning of the process, so that logic should exist elsewhere..probably in frontend?) 

 Order players are asked to disprove suggestion is the same as turn order..list was intended to be suitable for both of those uses. If we use a single queue and pop/push during suggestion resolution it'll screw up the movement order. Workaround would be to copy and generate a 2nd temp queue every time a suggestion is active or something..but idk if that's cleaner than just using a single list and indexing in. Depends where/how suggestions are handled, at which point this can be revisited.